pkg/operator/controller/credentials-secret/credentials_secret.go

```diff
@@ -124,10 +124,18 @@ func desiredCredentialsSecret(sourceSecret *corev1.Secret, destName types.Namesp
 		case operatorv1alpha1.ProviderTypeAWS:
 			secret.Data = sourceSecret.Data
 		}
-	} else {
-		secret.Data = sourceSecret.Data
+		return secret
+	}
+
+	if extDNS.Spec.Provider.Type == operatorv1alpha1.ProviderTypeAWS {
+		secret.StringData = make(map[string]string)
+		for key, value := range sourceSecret.Data {
+			secret.StringData[key] = string(value)
+		}
+		return secret
```

Comment:
This code is no different from the simple assignment which happens after: `secret.Data = sourceSecret.Data`.
What should happen here is the following:
- if the provider is AWS
  - If the source secret doesn't have the key `credentials`
      - take the value of the key `aws_access_key_id` from the source secret
      - take the value of the key `aws_secret_access_key` from the source secret
      - use [the function](https://github.com/openshift/cluster-ingress-operator/blob/58be189af289878b0c9b141df78814c08b83fcfa/pkg/util/aws/shared_credentials_file.go#L52-L58) to transform the just retrieved KeyID and AccessSecret into a credentials file on the fly
      - add `credentials` field into the target secret
   - Else: `secret.Data = sourceSecret.Data`
- Else: `secret.Data = sourceSecret.Data`

This would make sure that we always have `credentials` key in the target secret even if the users still used the API the old way (with the keyID and accesssecret fields).